### ISSUE TITLE
Implementing CI

### ISSUE THREAD

**Issue by okpatil4u** [reddit-india/indianelectionmeter.github.io#61]
Using travis-ci or circle-ci for bug free builds.

**Comment by jimaldon** [comment]
I have experience with jenkins, but not travis and circle.

**Comment by okpatil4u** [comment]
Anything could work. How do you set it up ?
Can you write few basic tests ? I

**Comment by ndarville** [comment]
I’ve got my own repo with custom rules and linters: https://github.com/ndarville/style. You can see an example of it in action at [another repo](https://github.com/hafniatimes/hafniatimes.github.io/blob/master/.travis.yml).

You’ll have to decide on the linter rules, if you don’t want to stick with the defaults.

**Comment by thel3l** [comment]
This is easy enough. 
First go to travis-ci.org  and link your github account ( the one which owns this repository).
Then enable the integration - screenshot attached. After this, anyone with push access can push the .Travis.yml file required for Travis to run - I'm willing to write a .Travis.yml file, but I'll need you to get set up the repo first. 
![screenshot_2015-11-22-07-11-24](https://cloud.githubusercontent.com/assets/9950250/11321513/7fc22b5a-90e8-11e5-98ea-dbab4b54b20e.png)

**Comment by ndarville** [comment]
Also check out [LintHub](https://linthub.io/). Really easy to enable, and it lints your pull requests.

**Comment by okpatil4u** [comment]
@ndarville @thel3l I am not looking for information. I am looking someone to take over this issue. Can any of you take it on ?

**Comment by thel3l** [comment]
Only the owner of this repo can set it up. None of us can.

**Comment by jimaldon** [comment]
I can take it up after you set it up.

**Comment by okpatil4u** [comment]
Thanks @jimdsouza. Should it be travis-ci or circle-ci ?

**Comment by jimaldon** [comment]
Whichever looks easily integrable. Since I'll have to do a lot of digging either way, I suggest you pick one with good documentation and a big community.

**Comment by thel3l** [comment]
I'd choose Travis.

**Comment by jimaldon** [comment]
Okay. 
Since you started this repo, @okpatil4u, could you go here (https://docs.travis-ci.com/user/getting-started/)  and complete steps 1 and 2? And then add me as a contributor to this repo?

**Comment by okpatil4u** [comment]
Done

**Comment by ndarville** [comment]
What exactly is it you want to be checked?

I [tried my hand at it](https://travis-ci.org/ndarville/indianelectionmeter.github.io), and it looks like Meteor.js can’t be checked directly with Travis; you’ll have to write a test suite for it instead, from what I gather. You can just run a meteor instance, because it’ll time out and end the test suite.

Some other things to test and lint would be:
- JS(X)
- HTML
- CSS
- JSON

The last one is the most straightforward, but again, it depends on what you want in a linter. I see that people are using very, very different coding styles in JS, and we need to agree on a standard, unless you just want me to enforce one (four-space indentation and so on).

---

The best linter to go by is `jscs`, because it supports automatic code fixes, even in editors like Sublime Text. This makes it a little easier to rewrite the code base as it currently exists.

I don’t have any JSX-specific linter rules, though, but there are still the basics.

**Comment by thel3l** [comment]
@ndarville I think it's possible. If @okpatil4u is done with the first two steps, then I'll try my hand at this.

**Comment by okpatil4u** [comment]
@thel3l, yeah, I have completed first two steps, but @jimdsouza is taking over it. It won't make sense all of us working on this one. I will try to resolve most of content generator issues by this monday. Why don't you take care of web scraper ?

**Comment by thel3l** [comment]
Ook. That's fine with me.

**Comment by okpatil4u** [comment]
@jimdsouza any progress ?

**Comment by jimaldon** [comment]
@okpatil4u In the thick of my exams this week. Plan to dive from Wednesday next week.

**Comment by okpatil4u** [comment]
Sounds good. I was also busy last week, finishing most of the tasks today.
